samcli/commands/build/click_container.py

```diff
@@ -0,0 +1,27 @@
+"""
+Module to check container based cli parameters
+"""
+import click
+
+
+class ContainerOptions(click.Option):
+    """
+    Preprocessing checks for presence of --use-container flag for container based options.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.require_container: bool = kwargs.pop("require_container", False)
```

Comment:
I am not sure why we need this `require_container`, as far as I can search from this PR change, `require_container` is always `True` when `ContainerOptions` is used. If we remove it, would it still work?

Reply:
in the future if we get more complex with conditions and dependencies we can continue using this class adding other variables, that's the main reason why we design it this way

Reply:
actually it kind of makes sense to remove this completely, for additional logic they would still require container as a prereq

Reply:
👍 thx for addressing